app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioGroup.kt

```diff
@@ -0,0 +1,905 @@
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioGroupTokens.LargeSizeRadioGroupTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioGroupTokens.MediumSizeRadioGroupTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioGroupTokens.SmallSizeRadioGroupTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioOptionTokens.LargeSizeRadioOptionTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioOptionTokens.MediumSizeRadioOptionTokens
+import com.hedvig.android.design.system.hedvig.tokens.SizeRadioOptionTokens.SmallSizeRadioOptionTokens
+import hedvig.resources.R
+
+@Composable
+fun RadioGroup(
+  data: List<OptionData>,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+  groupLockedState: LockedState = NotLocked,
+  radioGroupStyle: RadioGroupStyle = RadioGroupDefaults.radioGroupStyle,
+  radioGroupSize: RadioGroupSize = RadioGroupDefaults.radioGroupSize,
+) {
+  val contentPadding = calculateContentPadding(radioGroupStyle, radioGroupSize)
+  when (radioGroupStyle) {
+    Horizontal -> HorizontalRadioGroup(
+      data = data,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is HorizontalWithLabel -> HorizontalRadioGroupWithLabel(
+      data = data,
+      groupLabelText = radioGroupStyle.groupLabelText,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      contentPaddingValues = contentPadding,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is Vertical -> VerticalRadioGroup(
+      data = data,
+      radioGroupStyle = radioGroupStyle,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+
+    is VerticalWithGroupLabel -> VerticalRadioGroupWithLabel(
+      data = data,
+      radioGroupStyle = radioGroupStyle,
+      modifier = modifier,
+      groupLockedState = groupLockedState,
+      radioGroupSize = radioGroupSize,
+      contentPaddingValues = contentPadding,
+      onOptionClick = { id ->
+        onOptionClick(id)
+      },
+    )
+  }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+private fun HorizontalRadioGroup(
+  data: List<OptionData>,
+  groupLockedState: LockedState,
+  radioGroupSize: RadioGroupSize,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  FlowRow(
+    modifier,
+    maxItemsInEachRow = 2,
+    overflow = Companion.Visible,
+    verticalArrangement = Arrangement.spacedBy(4.dp, Alignment.CenterVertically),
+  ) {
+    for (i in data) {
+      val itemPadding = if (data.indexOf(i) % 2 != 0) {
+        PaddingValues()
+      } else {
+        PaddingValues(end = 4.dp)
```

Comment:
A bit confused here how there is a need for 4dp on the right only for the items that are on the right side in this flowrow, do the left items get this from somewhere else that I am not seeing?

Reply:
it's actually the other way round, with items on the left (index 0, 2 etc) they need a space between it and the right item, and items on the right (index 1,3) are at the end of two-items row, so they don't need extra padding there